src/main/kotlin/lotto/domain/WinningCombination.kt

```diff
@@ -0,0 +1,20 @@
+package lotto.domain
+
+import lotto.exceptions.LottoException
+import lotto.util.LottoUtils
+
+/**
+ * the class contain winning ticket and bonus number to eval user's Lotto tickets
+ */
+class WinningCombination(val winningTicket: Lotto, val bonusNumber: Int = 0) {
+    init {
+        LottoUtils.requireOrThrow(
+            LottoUtils.isInRange(bonusNumber),
+            LottoException.InvalidBonusNumberException(bonusNumber),
+        )
```

Comment:
```suggestion
        isInRange(bonusNumber).orThrow { LottoException.InvalidBonusNumberException(bonusNumber) }
```
One of Kotlin’s strengths is its ability to build DSLs and extend existing language features in a natural way.
With that in mind, we can actually refactor this logic to make it more expressive like the example above.

Let’s dive into Kotlin a bit and explore how we can achieve this!😄 
**Hint**: [extension function](https://kotlinlang.org/docs/extensions.html#extensions-are-resolved-statically)

Reply:
That's a great idea! I have to admit the concept of DSLs is still quite new to me, but I can already see how powerful it is.
It reminds me of operator overloading in C++, also powerful, but dangerous if misused. Without a clear purpose, it can add unnecessary complexity and hurt maintainability.

I'm especially still getting used to lambdas with receivers, they feel a bit tricky to me. Do you have any tips or mental models for understanding them better?

Back to the code, I agree, using an extension function here makes it much more expressive. What do you think about this version?
``` kotlin
fun Boolean.orThrow(lottoException: () -> LottoException) {
    if (!this) throw lottoException()
}

```
It feels clean and reusable.
Curious to hear your thoughts!👀

Reply:
Great.
As for the tip, for now I would recommend trying them out as we go through the missions.
I think in this point, searching and applying the knowledge in the project by him/herself
is the best and quickest way to understand.

Try to apply the knowledge in the project as much as possible.
I'll tell you if something goes wrong, or I feel that it's too far.